### PR TITLE
Fix autodetect

### PR DESCRIPTION
**Pull request by haslinghuis**
- Prevent auto-detect in manual or virtual mode.
- Moved auto-detect code out of the main loop

### CODE REVIEW

**Review by nerdCopter — APPROVED** [review]
* tested good. i prefer manually clicking anyway.
* disable virtual is good too. (until we get SITL :wink: )

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

**Review by ctzsnooze — APPROVED** [review]

**Pull request merged**
